src/AutoRest.CSharp/Mgmt/Output/MgmtLongRunningOperation.cs

```diff
@@ -0,0 +1,109 @@
+            }
+        }
+
+        private CSharpType[]? Arguments { get; }
+        public override CSharpType Type => new CSharpType(this, isValueType: TypeKind is TypeKind.Struct or TypeKind.Enum, arguments: Arguments);
+
+        public Type BaseType { get; }
+        public string WaitMethod { get; }
+        public Type OperationType { get; }
+        public Type ResponseType { get; }
+        public Type OperationOrResponseType { get; }
+
+        private ConstructorSignature? _mockingCtor;
+        public ConstructorSignature? MockingCtor => _mockingCtor ??= EnsureMockingCtor();
+        private ConstructorSignature? EnsureMockingCtor()
+        {
+            return new ConstructorSignature(
+                Name: Type.Name,
+                Description: $"Initializes a new instance of the <see cref=\"{Type.Name}\"/> class for mocking.",
+                Modifiers: "protected",
+                Parameters: new Parameter[0]);
+        }
+
+        private ConstructorSignature? _responseCtor;
+        public ConstructorSignature? ResponseCtor => _responseCtor ??= EnsureResponseCtor();
+        private ConstructorSignature? EnsureResponseCtor()
+        {
+            return new ConstructorSignature(
+                Name: Type.Name,
+                Description: null,
+                Modifiers: "internal",
+                Parameters: new[] { GenericResponseParameter });
+        }
+
+        private ConstructorSignature? _requestCtor;
+        public ConstructorSignature? RequestCtor => _requestCtor ??= EnsureRequestCtor();
+        private ConstructorSignature? EnsureRequestCtor()
+        {
+            var parameters = new List<Parameter> { ClientDiagnosticParameter, PipelineParameter, RequestParameter, ResponseParameter, FinalStateParameter };
+            if (SourceParameter is not null)
+                parameters.Insert(0, SourceParameter);
+            return new ConstructorSignature(
+                Name: Type.Name,
+                Description: null,
+                Modifiers: "internal",
+                Parameters: parameters.ToArray());
+        }
+
+        protected Parameter GenericResponseParameter => new Parameter("response", null, ResponseType, null, false);
+
+        protected Parameter? SourceParameter => IsGeneric ? new Parameter("source", null, typeof(IOperationSource<>), null, false) : null;
+        protected Parameter ClientDiagnosticParameter => new Parameter("clientDiagnostics", null, typeof(ClientDiagnostics), null, false);
+        protected Parameter PipelineParameter => new Parameter("pipeline", null, typeof(HttpPipeline), null, false);
+        protected Parameter RequestParameter => new Parameter("request", null, typeof(Request), null, false);
+        protected Parameter ResponseParameter => new Parameter("response", null, typeof(Response), null, false);
+        protected Parameter FinalStateParameter => new Parameter("finalStateVia", null, typeof(OperationFinalStateVia), null, false);
```

Comment:
`MgmtLongRunningOperation` has no derived types, so there properties can be made private. Also, they can be static readonly, similar to https://github.com/Azure/autorest.csharp/blob/feature/v3/src/AutoRest.CSharp/Common/Output/Models/Shared/KnownParameters.cs (parameter instances can be shared between method signatures).